sdk/servicebus/azure-servicebus/azure/servicebus/aio/_servicebus_sender_async.py

```diff
@@ -458,14 +460,19 @@ async def create_message_batch(self, max_size_in_bytes: Optional[int] = None) ->
         if not self._max_message_size_on_link:
             await self._open_with_retry()
 
-        if max_size_in_bytes and max_size_in_bytes > self._max_message_size_on_link:
+        if self._max_message_size_on_link > MAX_BATCH_SIZE_PREMIUM:
+            max_size_allowed = MAX_BATCH_SIZE_PREMIUM
```

Comment:
lets make `max_size_allowed` be something like  `self._max_batch_size_on_link` and populated right after here https://github.com/Azure/azure-sdk-for-python/blob/480ed8878e88aba478ed6132def8d51bdf0cc909/sdk/eventhub/azure-eventhub/azure/eventhub/_client_base.py#L549 . This way if the metadata from the service ever returns the right size itll be a cleaner update.